nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-value-providers/src/main/java/org/apache/nifi/stateless/parameter/aws/SecretsManagerParameterValueProvider.java

```diff
@@ -0,0 +1,208 @@
+            }
+        }
+    }
+
+    @Override
+    public boolean isParameterDefined(final String contextName, final String parameterName) {
+        return supportedParameterNames.contains(getSecretName(contextName, parameterName));
+    }
+
+    @Override
+    public String getParameterValue(final String contextName, final String parameterName) {
+        final String secretName = getSecretName(contextName, parameterName);
+        final GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest()
+                .withSecretId(secretName);
+        final GetSecretValueResult getSecretValueResult = secretsManager.getSecretValue(getSecretValueRequest);
+
+        if (getSecretValueResult == null) {
+            throw new IllegalArgumentException(String.format("Secret [%s] not found", secretName));
+        }
+        if (getSecretValueResult.getSecretString() != null) {
+            return getSecretValueResult.getSecretString();
+        } else {
+            throw new IllegalStateException("Binary secrets are not supported");
+        }
+    }
+
+    private Properties loadProperties(final String propertiesFilename) throws IOException {
+        final Properties properties = new Properties();
+
+        try (final InputStream in = new FileInputStream(Paths.get(propertiesFilename).toFile())) {
+            properties.load(in);
+            return properties;
+        }
+    }
+
+    AWSSecretsManager configureClient(final String awsBootstrapConfFilename) throws IOException {
+        if (awsBootstrapConfFilename == null) {
+            return getDefaultClient();
+        }
+        final Properties properties = loadProperties(awsBootstrapConfFilename);
+        final String accessKey = properties.getProperty(ACCESS_KEY_PROPS_NAME);
+        final String secretKey = properties.getProperty(SECRET_KEY_PROPS_NAME);
+        final String region = properties.getProperty(REGION_KEY_PROPS_NAME);
+
+        if (isNotBlank(accessKey) && isNotBlank(secretKey) && isNotBlank(region)) {
+            return AWSSecretsManagerClientBuilder.standard()
+                    .withRegion(region)
+                    .withCredentials(new BasicAWSCredentialsProvider(accessKey, secretKey))
+                    .build();
+        } else {
+            return getDefaultClient();
+        }
+    }
+
+    private AWSSecretsManager getDefaultClient() {
+        return AWSSecretsManagerClientBuilder.standard()
+                .withCredentials(DefaultAWSCredentialsProviderChain.getInstance())
+                .build();
+    }
+
+    private static String getSecretName(final String contextName, final String parameterName) {
+        return String.format(QUALIFIED_SECRET_FORMAT, contextName, parameterName);
+    }
+
+    private static boolean isNotBlank(final String value) {
+        return value != null && !value.trim().equals("");
+    }
+
+    private static class BasicAWSCredentialsProvider implements AWSCredentialsProvider {
```

Comment:
Is this class necessary?  It seems like the standard [BasicAWSCredentials](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/BasicAWSCredentials.html) class should work.

Reply:
Good catch with `BasicAWSCredentials`, I missed that class.  Now, the `AwsClientBuilder#withCredentials` (of which `AWSSecretsManagerClientBuilder` is a subclass) requires an `AWSCredentialsProvider` rather than an `AWSCredentials`, but `BasicAWSCredentials` would definitely be preferable to the anonymous class currently returned by the `getCredentials` method.

Reply:
[AWSStaticCredentialsProvider](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSStaticCredentialsProvider.html) should work the `AWSCredentialsProvider`.

Reply:
Ah nice, that's exactly what I want.  Thanks!